includes/class-edd-download.php

```diff
@@ -630,4 +641,31 @@ public function is_free( $price_id = false ) {
 
 	}
 
+	/**
+	 * Determine if the download supports a feature
+	 * 
+	 * @since  2.4
+	 * @param  string $feature   The feature being checked for support
+	 * @return bool              Whether the download supports the feature.
+	 */
+	public function supports( $features = '' ) {
+		$supports = apply_filters( 'edd_download_supports', $this->supports, $this->ID );
```

Comment:
Assuming we set it up that way, how would an extension add extra features? Unless we make it a static/global set of download supported features. Maybe make it a static $supports array variable, and static add_support_feature/remove_support_feature method as well to modify that array.

Then we just pull that static array into the support() method. This would more closely resemble the add/remove_post_type_support functions you mentioned before.